packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -95,6 +100,23 @@ export default util.createRule<Options, MessageId>({
             context.report({
               node,
               messageId: 'floating',
+              suggest: [
+                {
+                  messageId: 'floatingFixAwait',
+                  fix(fixer): TSESLint.RuleFix {
+                    if (
+                      expression.type === AST_NODE_TYPES.UnaryExpression &&
+                      expression.operator === 'void'
+                    ) {
+                      return fixer.replaceTextRange(
+                        [expression.range[0], expression.range[0] + 4],
+                        'await',
+                      );
+                    }
+                    return fixer.insertTextBefore(node, 'await ');
+                  },
```

Comment:
this fixer isn't currently completely safe.
you'll need to check the precedence of the node to ensure that it doesn't need parentheses.

[we have a util for this](https://github.com/typescript-eslint/typescript-eslint/blob/530e0e618cdf4bb956149bf8a8484848e1b9a1f5/packages/eslint-plugin/src/util/getOperatorPrecedence.ts#L195) - you can find usages in the codebase for examples!

Reply:
some examples:
```ts
declare function floatingPromise(): Promise<void>;
declare const bool: boolean;

async function *main() {
  floatingPromise();
  // fix to
  await floatingPromise();

  // example that NEEDS parentheses
  1, floatingPromise();
  // if you don't add parentheses, then you have bad code!!!
  await 1, floatingPromise();
  // correct fix with parentheses
  await (1, floatingPromise());


  // another example
  bool ? floatingPromise() : null;
  // without parentheses - broken!!
  await bool ? floatingPromise() : null;
  // with parentheses - fine
  await (bool ? floatingPromise() : null);
}
```
[playground](https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=CYUwxgNghgTiAEAzArgOzAFwJYHtVIhym1QHMAFGHAWywGcQAKASgC55Kb6QAeANxxZgAPgDcAKFCRYCMHjoZ4AIxw4I7FWpBRUE8VDoBPdEjSZc+AFTUoWVC3gBvcfAJESFKrQYsJrgPT+SFgAHvAYOC7wUADutoqIhMR2nlw+zHoBQSAhUNQADhAIGAAWxPAAcgCiVQAiAMrw+TKopSAMdFEAjAA0bslknN5MGVGB8FiI8IY4yPDAeADkilDAwE0tbR19bfgzc2V8CEqr8HKgAIRXUbHx8L39HkPcvmNBcjBwmMFhMVilGzgrRK7XaNzi-3gjAeiXcKWe6VG4je0VQODaMHgOTyhRAUU0EHgAH5HvCvC9mPB2KhkBAIH54OM-qVZopmkCtu14ABaZRUADWIFQ11ct0hBOJpMG5PSVPgNLpDKZ-xKgKFnLoPOCqDxoohikYEpJsIGqWGDmptIgowAvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTooiaBOiRwYAL4h1QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)

Reply:
@bradzacher 
Thank you.
I have change the code that check the precedence of the node .
And add tests that fit your examples.
Is this okay?

1ddb23b5f3e731808b3928836d1db30ad9965c5d